chipy_org/apps/meetings/models.py

```diff
@@ -276,17 +331,85 @@ def clean(self):
     def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
         self.full_clean()
 
+        original = None
+        if self.id:
+            original = RSVP.objects.get(id=self.id)
+
         # Generate a key for this RSVP
         if not self.key:
             self.key = "".join(
                 random.choice(string.digits + string.ascii_lowercase) for x in range(40)
             )
 
-        return super(RSVP, self).save(*args, **kwargs)
+        if not (original and original.response == self.response):
+            if self.response == self.Responses.IN_PERSON:
+                if self.meeting.has_in_person_capacity():
+                    self.status = self.Statuses.CONFIRMED
+                else:
+                    self.status = self.Statuses.WAIT_LISTED
+
+            if self.response == self.Responses.VIRTUAL:
+                if self.meeting.has_virtual_capacity():
+                    self.status = self.Statuses.CONFIRMED
+                else:
+                    self.status = self.Statuses.WAIT_LISTED
+
+        if self.response == self.Responses.DECLILNED:
+            self.status = self.Statuses.CONFIRMED
+
+        return super().save(*args, **kwargs)
 
     @property
     def full_name(self):
         return self.first_name + " " + self.last_name
 
     def __str__(self):
         return f"{self.meeting}: {self.full_name}"
+
+
+@receiver(post_save, sender=RSVP)
+def rsvp_post_save(sender, instance, **kwargs):
```

Comment:
This method is a great case for us getting that background worker in place. Imagining a case where 100 people were on the backlog post resize could trigger 99 queries and 99 saves and 99 emails in serial.